test/python/test_pool.py

```diff
@@ -0,0 +1,164 @@
+
```

Comment:
IMHO it probably better to put this in a subdir called pool, with its own dockerfile perhaps?

Reply:
Yes I think this should happen eventually, but I also think there might be scope for further reorganisation as well.